### ISSUE TITLE
accidentally deleted @hostmaster... any way to get it back?

### ISSUE THREAD

**Issue by heliogabal** [omega8cc/boa#632]
I did a very stupid thing and accidentally deleted my @hostmaster site on my development-Server with `drush @hostmaster provision-delete @platform_XXX` (obviously thinking I would delete @platform_XXX with it). Now it seems this did a backup to @platform_XXX in /data/disk/o1/clients/admin/backups/, which was automatically deleted in the process, so I can't find a backup anywhere. All I have is an older *tar.gz-backup, but I'm afraid I will loose data if I recover that.

Is there an easy way to recreate the @hostmaster platform without loosing all its data, like simply re-verifiying or redoing octopus-install? If not, how could I best create a new octopus instance in this case and migrate the orphaned platforms to it?  

Thanks for helping out here, I'm anxious to touch anything before I get an informed opinion on this as to not make things any worse (which I feel is barely possible)...

**Comment by omega8cc** [comment]
This is rather serious mistake. If you don't know how to re-create the hostmaster site and its drush aliases, using db backup from `/data/disk/arch/sql/` then you should probably just install another octopus and move platforms with sites there.

The steps would be (with example where o1 is old and o2 is new) per platform (make sure to use correct paths)

`$ boa in-octopus your@email o2 none`
`$ mv /data/disk/o1/static/my-platform /data/disk/o2/static/
`$ chown -R o2 /data/disk/o2/static/*`

Add that platform in o2 hostmaster so it will import all sites, and only then move away duplicate vhosts from o1:

`$ mv /data/disk/o1/config/server_master/nginx/vhost.d/* /var/backups/`
`$ service nginx reload`

Now you may need to run registry rebuild task on all sites on o2.

That should do the trick.

**Comment by heliogabal** [comment]
I know, I know, I like to learn the hard way... but at least, mistakes like that you do only once. 

Thanks a lot for pointing out the best procedure to fix my mess, once again you guys saved my day. One minor change I had to make: instead of `$ boa in-octopus your@email o2 none` I had to use `$ boa in-stable your@email o2 none`, then all the rest works like a charm. Just in case someone else needs this one day...
